clifford/_mvarray.py

```diff
@@ -9,14 +9,38 @@
 call_array = np.vectorize(MultiVector.__call__)
 
 
+def _interrogate_nested_mvs(input_array):
+    """
+    Calculates the shape of the nested input_array, and gets the associated layout.
+    Stops descending when it encounters a MultiVector.
+    """
+    if not isinstance(input_array[0], MultiVector):
+        nested_shape, layout, dtype = _interrogate_nested_mvs(input_array[0])
+        return (len(input_array), *nested_shape), layout, dtype
+    else:
+        return tuple([len(input_array)]), input_array[0].layout, type(input_array[0].value[0])
+
+
+def _index_nested_iterable(input_iterable, index):
+    """
+    Given a nested iterable, input_iterable, return the element given by the
+    1d index iterable
+    """
+    res = input_iterable[index[0]]
+    for ind in index[1:]:
+        res = res[ind]
+    return res
```

Comment:
Again, can move the base-case back:
```suggestion
    res = input_iterable
    for ind in index:
        res = res[ind]
    return res
```